rust/agama-cli/src/auth.rs

```diff
@@ -37,15 +38,35 @@ pub async fn run(subcommand: AuthCommands) -> anyhow::Result<()> {
 fn read_password() -> Result<String, CliError> {
     let stdin = io::stdin();
     let password = if stdin.is_terminal() {
-        rpassword::prompt_password("Please, introduce the root password: ")?
+        ask_password()?
     } else {
         let mut buffer = String::new();
-        stdin.read_line(&mut buffer)?;
+        stdin
+            .read_line(&mut buffer)
+            .map_err(|_| CliError::MissingPassword)?;
         buffer
     };
     Ok(password)
 }
 
+/// Asks interactively for the password.
+fn ask_password() -> Result<String, CliError> {
+    let validator = |input: &str| {
+        if input.is_empty() {
+            Ok(Validation::Invalid("The password cannot be blank.".into()))
```

Comment:
This is reasonable for *setting* a password (along with more nontrivial validations).
I don't see the purpose for this, for *verifying* a password.

Reply:
We are just making sure that the password is not empty.